pkg/api/image.go

```diff
@@ -51,41 +51,53 @@ func (h *Handler) GetImages(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		// get all the image names for the current pool
-		imageNames, err := ioctx.GetImageNames()
-		if err != nil {
-			logger.Errorf("failed to get image names from pool %s: %+v", p.Name, err)
-			w.WriteHeader(http.StatusInternalServerError)
+		images, ok := getImagesForPool(w, p.Name, ioctx)
+		if !ok {
 			return
 		}
 
-		// for each image name, open the image and stat it for further details
-		images := make([]model.BlockImage, len(imageNames))
-		for i, name := range imageNames {
-			image := ioctx.GetImage(name)
-			image.Open(true)
-			defer image.Close()
-			imageStat, err := image.Stat()
-			if err != nil {
-				logger.Errorf("failed to stat image %s from pool %s: %+v", name, p.Name, err)
-				w.WriteHeader(http.StatusInternalServerError)
-				return
-			}
-
-			// add the current image's details to the result set
-			images[i] = model.BlockImage{
-				Name:     name,
-				PoolName: p.Name,
-				Size:     imageStat.Size,
-			}
-		}
-
 		result = append(result, images...)
 	}
 
 	FormatJsonResponse(w, result)
 }
 
+func getImagesForPool(w http.ResponseWriter, poolName string, ioctx ceph.IOContext) ([]model.BlockImage, bool) {
+	// ensure the IOContext is destroyed at the end of this function
+	defer ioctx.Destroy()
+
+	// get all the image names for the current pool
+	imageNames, err := ioctx.GetImageNames()
+	if err != nil {
+		logger.Errorf("failed to get image names from pool %s: %+v", poolName, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return nil, false
```

Comment:
You are logging the error but should you pass it back to the caller? so that it is returned to the user ?